server.go

```diff
@@ -140,8 +142,9 @@ func (server *mongoServer) AcquireSocket(poolLimit int, minPoolSize int, timeout
 			socket = &mongoSocket{
 				socketState: Connecting,
 			}
-			// hold our spot in the liveSockets slice
-			server.liveSockets = append(server.liveSockets, socket)
+			// hold a spot in the liveSockets slice to ensure connecting sockets are counted
```

Comment:
Can you explain as to why is this important and a close on a socket no longer needed (even if over the pool limit will be detrimental)? This is the thundering herd problem I am assuming (where we run out of connections), so might be a good comment to leave here for future readers.

Reply:
Pushing a comment explaining the pool limit on the `mongoCluster` socket, and adjusting terminology here to more obviously related it to the `poolLimit` parameter which can be traced back to `mongoCluster`.

In case it matters to anyone in the future, [here](https://github.com/lyft/mgo/commit/61b7e276ad0ac5597c5a154d621be6caba6ccf2d#diff-34c6b408d72845d076d47126c29948d1R140)'s the relevant change that changed how pool limits work.